cpp/src/merge/merge.cu

```diff
@@ -418,8 +462,15 @@ table_ptr_type merge(cudf::table_view const& left_table,
 
   // extract merged row order according to indices:
   //
-  auto const merged_indices = generate_merged_indices(
-    index_left_view, index_right_view, column_order, null_precedence, nullable, stream);
+  index_vector merged_indices(0, stream);
+  if (cudf::detail::has_nested_columns(left_table) or
+      cudf::detail::has_nested_columns(right_table)) {
+    merged_indices = generate_merged_indices_nested(
+      index_left_view, index_right_view, column_order, null_precedence, nullable, stream);
+  } else {
+    merged_indices = generate_merged_indices(
+      index_left_view, index_right_view, column_order, null_precedence, nullable, stream);
+  }
```

Comment:
Use an IIFE to avoid requiring a stream-ordered constructor. This also lets us mark the result as `const`.
```suggestion
  auto const merged_indices = [&](){
    if (cudf::detail::has_nested_columns(left_table) or
        cudf::detail::has_nested_columns(right_table)) {
      return generate_merged_indices_nested(
        index_left_view, index_right_view, column_order, null_precedence, nullable, stream);
    } else {
      return generate_merged_indices(
        index_left_view, index_right_view, column_order, null_precedence, nullable, stream);
    }
  }();
```